OSDOCS-18782 adding ESO docs to 4.19 for GA (#109778)
openshift/openshift-docs: modules/external-secrets-enable-operator-log-level.adoc

// Module included in the following assemblies:
//
// * security/external_secrets_operator/external-secrets-log-levels.adoc

:_mod-docs-content-type: PROCEDURE
[id="external-secrets-enable-operator-log-level_{context}"]
= Setting a log level for the {external-secrets-operator}

[role="_abstract"]

> Set the log level for the {external-secrets-operator} to control the detail of log messages. By adjusting the verbosity, you can troubleshoot issues effectively and manage the volume of log data.

Reviewer: ```suggestion
You can adjust the verbosity of your logs to both troubleshoot issues effectively and manage the volume of log data. Set the log level for the {external-secrets-operator} to control the detail of log messages.
```

Author: Done

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.
* You have created the `ExternalSecretsConfig` custom resource.

.Procedure

> * Update the subscription object for the {external-secrets-operator} to provide the verbosity level for the operator logs by running the following command:

Author: Done

> * Update the subscription object for the {external-secrets-operator} to provide the verbosity level for the operator logs by running the following command:

Author: Done

+
[source,terminal]
----
$ oc -n <external_secrets_operator_namespace> patch subscription openshift-external-secrets-operator --type='merge' -p '{"spec":{"config":{"env":[{"name":"OPERATOR_LOG_LEVEL","value":"<log_level>"}]}}}'
----
+
where:

> external_secrets_operator_namespace:: Specifies the namespace where the Operator is installed.

Reviewer: ```suggestion
<external_secrets_operator_namespace>:: Specifies the namespace where the Operator is installed.
```
user-replaced values in angle brackets (see SSG)

Author: Done

> log_level:: Specifies the level of log detail. Values range from 1-5. The default is 2.

Reviewer: ```suggestion
<log_level>:: Specifies the level of log detail. Values range from 1-5. The default is 2.
```

Author: Done

.Verification

> . The External Secrets Operator pod is redeployed. Verify that the log level of the {external-secrets-operator} is updated by running the following command:

Reviewer: ```suggestion
. Verify that the log level of the {external-secrets-operator} is updated by running the following command:
```

Author: Done

+
[source,terminal]
----
$ oc set env deploy/external-secrets-operator-controller-manager -n external-secrets-operator --list | grep -e OPERATOR_LOG_LEVEL -e container
----
+

> The following example verifies that the log level of the {external-secrets-operator} is updated.

Reviewer: ```suggestion
```

Author: Done

> +

Reviewer: ```suggestion
+.Example output
```

Author: Done

[source,terminal]
----
# deployments/external-secrets-operator-controller-manager, container manager
OPERATOR_LOG_LEVEL=2
----

> . Verify that the log level of the {external-secrets-operator} is updated by running the `oc logs` command:

Reviewer: ```suggestion
. Verify that the log level of the {external-secrets-operator} is updated by running the following command:
```
you will be annoyed if this command changes; recommend keeping the step generic (always)

Author: Done
